src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,52 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public final String value;
+    protected boolean isPrivate;
+    
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate, String validationRegex, String constraints) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        contact = contact.trim();
+        if (!isValid(contact, validationRegex)) {
+            throw new IllegalValueException(constraints);
+        }
+        this.value = contact;
+    }
+
+    /**
+     * Checks if a given string is a valid person contact.
+     */
+    public static boolean isValid(String test, String validationRegex) {
+        return test.matches(validationRegex);
+    }
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
+    }
```

Comment:
Be careful, since you can compare different objects here, such as comparing `Address` object with `Email` object.